io/pci/EEH.py

```diff
@@ -121,31 +129,41 @@ def test_eeh_basic_pe(self):
                                                      self.add_cmd)
                     if return_code == EEH_MISS:
                         num_of_miss += 1
-                        self.log.info("number of miss is %d", num_of_miss)
+                        self.log.info(f"number of miss is {num_of_miss}")
                         continue
                     else:
                         num_of_hit += 1
                         self.log.info("number of hit is %d", num_of_hit)
                         if num_of_hit <= self.max_freeze:
                             if not self.check_eeh_pe_recovery():
-                                self.fail("PE %s recovery failed after"
-                                          "%d EEH" % (self.pci_device,
-                                                      num_of_hit))
+                                self.fail(
+                                        f"PE {self.pci_device} recovery failed after {num_of_hit}"
+                                        f" EEH"
+                                )
                                 break
                             else:
                                 self.log.info("PE recovered successfully")
+                        time.sleep(10)
+                        if pci.get_pci_class_name(self.pci_device) == "fc_host":
+                            after_eeh_path_status = multipath.get_multipath_details()
+                            get_diff_bef_aft = genericutility.recursive_compare_dict(
+                                    before_eeh_path_status, after_eeh_path_status
```

Comment:
@maramsmurthy some how I am not convinced with dictionary way of validation, because the dictionary holds many different parameters which may change after driver reload on EEH.
if any of the parameter also gets changed then the test will fail.

So my view is just focus on only what needs to be validated than validating everything which is even not necessary.

Reply:
@Naresh-ibm You are correct. I verified with successfull EEH recovery case by doing same comparison. Only two attributes are getting impacted as per your "switch_grp" and "path_faults". I will be excluding these attributes from comparison.

Reply:
@Naresh-ibm  Addressed the problem as suggested

Reply:
Please find attached log file for reference
[job_dict_check.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/11207825/job_dict_check.log)